Conference-Android/app/src/main/java/com/sagetech/conference_android/app/ui/activities/ConferenceListActivity.java

```diff
@@ -0,0 +1,142 @@
+package com.sagetech.conference_android.app.ui.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.sagetech.conference_android.app.R;
+import com.sagetech.conference_android.app.api.ConferenceController;
+import com.sagetech.conference_android.app.model.ConferenceData;
+import com.sagetech.conference_android.app.ui.listener.RecyclerItemClickListener;
+import com.sagetech.conference_android.app.ui.presenter.ConferenceListActivityPresenter;
+import com.sagetech.conference_android.app.ui.presenter.IConferenceListActivity;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import timber.log.Timber;
+
+public class ConferenceListActivity extends InjectableActionBarActivity implements IConferenceListActivity {
+
+    @Inject
+    ConferenceController conferenceController;
+
+    @InjectView(R.id.confView)
+    RecyclerView mRecyclerView;
+
+    private RecyclerView.Adapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+
+    private ConferenceListActivityPresenter presenter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_conferences);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(false);
+
+        ButterKnife.inject(this);
+
+        mRecyclerView.setHasFixedSize(true);
+
+        mLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(mLayoutManager);
+
+        mRecyclerView.addOnItemTouchListener(
+                new RecyclerItemClickListener(this, new RecyclerItemClickListener.OnItemClickListener() {
+                    @Override public void onItemClick(View view, int position) {
+
+                        Timber.d(String.format("View Clicked: %s", view.getTag()));
+
+                        Intent conferenceSessionListIntent = new Intent(view.getContext(), ConferenceSessionListActivity.class);
+                        conferenceSessionListIntent.putExtra("id", (Integer) view.getTag());
+                        startActivity(conferenceSessionListIntent);
+                    }
+                })
+        );
```

Comment:
I'm not sure I'm a big fan of this pattern for a RecyclerView. Recycler no longer supports the `OnItemClick` pattern for a number of reasons. I think it would be better to pass an `OnItemClickListener` to a property on the adapter, and set that listener on the view holder's `itemView`.

I would add a `setItemOnClickListener(View.OnClickListener listener)` method on the adapter, and in your `onBindViewHolder` set the `holder.itemView.setOnClickListener(mItemOnClickListener)`. I think Google (and if so, I agree) saw that the adapter is the handler of the view logic, while the `RecyclerView` should just be in charge of presenting what the adapter gives it.

Furthermore, this pattern would allow you to add multiple click listeners in a single item view if we decided, for instance, to turn these into `CardView`s with extra actions.